pkg/build/builder/sti.go

```diff
@@ -343,11 +344,11 @@ func (d *downloader) Download(config *s2iapi.Config) (*s2iapi.SourceInfo, error)
 // 2. In case of repeated Keys, the last Value takes precedence right here,
 //    instead of deferring what to do with repeated environment variables to the
 //    Docker runtime.
-func buildEnvVars(build *api.Build) map[string]string {
+func buildEnvVars(build *api.Build) s2iapi.EnvironmentList {
 	bi := buildInfo(build)
-	envVars := make(map[string]string, len(bi))
+	envVars := s2iapi.EnvironmentList{}
 	for _, item := range bi {
-		envVars[item.Key] = item.Value
+		envVars = append(envVars, s2iapi.EnvironmentSpec{Name: item.Key, Value: item.Value})
```

Comment:
you can also use `err = envVars.Set("KEY=VALUE")` to make sure the checks for the format/etc apply, but append will work as well.